tools/analyzers/longfile/longfile.go

```diff
@@ -2,53 +2,114 @@ package longfile
 
 import (
 	_ "embed"
+	"flag"
 	"fmt"
 	"go/ast"
 	"go/token"
+	"strconv"
 	"strings"
 
 	"golang.org/x/tools/go/analysis"
 )
 
-var max int
-var exceptions map[string]int
+var max = 1500
 
-var Analyzer = &analysis.Analyzer{
-	Name: "longfile",
-	Doc:  "detects if source code files are too long",
-	Run:  checkPath,
+const (
+	exceptionDoc = "comma separated list of files with size greater than the default." +
+		" each item in the list is with this format: <file name>:<number of lines>"
+)
+
+var Analyzer = newAnalyzer()
+
+func newAnalyzer() *analysis.Analyzer {
+	l := &longFileCfg{
+		// todo: once in rules_go version v0.32.0 or higher, nogo should send the command
+		//       line arguments (analyzer_flags field in nogo_configuration.json);
+		//       then we can replace this initialization by: `exceptions: make(longFileExceptions)`
+		// todo: try to reduce the size of each one of these files
+		exceptions: longFileExceptions{
+			"tests/storage/restore.go":        1621,
+			"tests/infra_test.go":             1686,
+			"tests/migration_test.go":         4382,
+			"tests/operator_test.go":          2990,
+			"tests/utils.go":                  2776,
```

Comment:
+1 for this; Would you drop my one-liner from `Makefile`? We do not want to tests for the same thing.

Reply:
Thanks @dankenigsberg. Done.